packages/eas-cli/src/credentials/ios/appstore/ensureAppExists.ts

```diff
@@ -1,24 +1,78 @@
+import { BundleId, CapabilityType, CapabilityTypeOption, Session, Teams } from '@expo/apple-utils';
 import ora from 'ora';
 
-import { AuthCtx } from './authenticate';
+import { AuthCtx, authenticateAsync } from './authenticate';
+import { USE_APPLE_UTILS } from './experimental';
 import { runActionAsync, travelingFastlane } from './fastlane';
 
+export interface EnsureAppExistsOptions {
+  enablePushNotifications?: boolean;
+}
+
 export interface AppLookupParams {
   accountName: string;
   projectName: string;
   bundleIdentifier: string;
 }
 
-export interface EnsureAppExistsOptions {
-  enablePushNotifications?: boolean;
+export async function ensureAuthenticatedAsync(
+  appleCtx: Omit<AuthCtx, 'fastlaneSession'>
+): Promise<Omit<AuthCtx, 'fastlaneSession'>> {
+  if (!Session.getSessionInfo()) {
+    appleCtx = await authenticateAsync({
+      appleId: appleCtx.appleId,
+      teamId: appleCtx.team.id,
+    });
+  }
+  Teams.setSelectedTeamId(appleCtx.team.id);
+  return appleCtx;
+}
```

Comment:
Hmmm, I feel this function doesn't belong to this file.
@wkozyra95 what do you think?

Reply:
I agree

I'mnjot really sure what is the goal here, with partial implementation for those changes to work user would need to login twice to both apis. If that that is intedended behaviour for now i would suggest to:
- move this function to `authenticate.ts`
- move all calls to this function to AppStoreApi.ts (depending on what function is used trigger correct ensureAuth)
- Make fastlaneSession optional in AuthCtx instead of `Omit<AuthCtx, 'fastlaneSession'>`